src/datetime_rotation.jl

```diff
@@ -0,0 +1,118 @@
+using Dates
+import Base: isless
+
+raw"""
+    DatetimeRotatingFileLogger
+
+Constructs a FileLogger that rotates its file based on the current date.
+The filename pattern given is interpreted through the `Dates.format()` string formatter,
+allowing for yearly all the way down to millisecond-level log rotation.  Note that if you
+wish to have a filename portion that is not interpreted as a format string, you may need
+to escape portions of the filename, as shown below:
+
+Usage example:
+
+    logger = DatetimeRotatingFileLogger(log_dir, raw"\a\c\c\e\s\s-YYYY-mm-dd.\l\o\g")
```

Comment:
I wonder if we should be used the format that Cron uses to specify periodic intervals?

Reply:
As stated below, (but I'll reiterate here for clarity) I think the main benefit to using a `DateFormat` is that it's a julian interface and allows people to very quickly define a filename that changes on a daily, hourly, monthly, etc... basis.  Which I think will cover 99% of all use cases.

If we don't want to use a `DateFormat`, I'd be more inclined to provide a functional interface, e.g. force the user to provide a function that returns `(time_to_next_rollover, filename)` when provided a `DateTime`.

Reply:
Yeah, I think its fine